pkg/apis/v1/labels.go

```diff
@@ -135,13 +129,21 @@ var (
 	}
 )
 
-// IsRestrictedLabel returns an error if the label is restricted.
+// IsRestrictedLabel is used for runtime validation of requirements.
+// Returns an error if the label is restricted. E.g. using .karpenter.sh suffix.
 func IsRestrictedLabel(key string) error {
 	if WellKnownLabels.Has(key) {
 		return nil
 	}
-	if IsRestrictedNodeLabel(key) {
-		return serrors.Wrap(fmt.Errorf("label is restricted; specify a well known label or a custom label that does not use a restricted domain"), "label", key, "well-known-labels", sets.List(WellKnownLabels), "restricted-labels", sets.List(RestrictedLabelDomains))
+	labelDomain := GetLabelDomain(key)
+	for restrictedLabelDomain := range RestrictedLabelDomains {
+		if labelDomain == restrictedLabelDomain || strings.HasSuffix(labelDomain, "."+restrictedLabelDomain) {
+			return fmt.Errorf("using label %s is not allowed as it might interfere with the internal provisioning logic; specify a well known label: %v, or a custom label that does not use a restricted domain: %v", key, sets.List(WellKnownLabels), sets.List(RestrictedLabelDomains))
```

Comment:
We should continue to use the serrror formatting from the source, I think we migrated to using structured errors after the original PR was created.